cranelift/filetests/src/test_run.rs

```diff
@@ -36,21 +36,26 @@ impl SubTest for TestRun {
     }
 
     fn run(&self, func: Cow<ir::Function>, context: &Context) -> SubtestResult<()> {
-        let mut compiler = SingleFunctionCompiler::with_host_isa(context.flags.clone());
+        // If this test requests to run on a completely different architecture than the host
+        // platform then we skip it entirely, since we won't be able to natively execute machine
+        // code.
+        let requested_arch = context.isa.unwrap().triple().architecture;
+        if requested_arch != Architecture::host() {
+            return Ok(());
+        }
```

Comment:
We could fix #1558 while we're at it: we probably want to print something to tell the user we ignored a run test because of hardware issues.